File: sdk/js/PicarioXPO.RenderAPI.V2/UrlGenerators/XpoUrlObject.ts
class XpoUrlObject {
    private index: number;
    private objectType: UrlGeneratorModule.XpoUrlObjectTypes;
    private text: XpoUrlText;
    private color: XpoUrlColor;
    private design: XpoUrlDesign;
    
    /*
     * Gets or sets the index of this overlay
     */
    getIndex(): number { return this.index; }
    setIndex(val: number) { this.index = val; }

    /*
     * Gets or sets the type for this object
     */
    getObjectType(): UrlGeneratorModule.XpoUrlObjectTypes { return this.objectType; }
    setObjectType(val: UrlGeneratorModule.XpoUrlObjectTypes) { this.objectType = val; }

    /*
     * Gets or sets the text for this object
     */
    getText(): XpoUrlText { return this.text; }
    setText(val: XpoUrlText) { this.text = val; }

    /*
     * Gets or sets the color for this object
     */
    getColor(): XpoUrlColor { return this.color; }
    setColor(val: XpoUrlColor) { this.color = val; }

    /*
     * Gets or sets the design for this object
     */
    getDesign(): XpoUrlDesign { return this.design; }
    setDesign(val: XpoUrlDesign) { this.design = val; }
} 

File: sdk/c#/PicarioXPO.RenderAPI/XpoUrlObject.cs
namespace PicarioXPO.RenderAPI
{
    /// <summary>
    /// Represents an object from a Picario scene for the XPO URL generator
    /// </summary>
    public sealed class XpoUrlObject
    {
        /// <summary>
        /// Gets or sets the index of this overlay
        /// </summary>
        public int Index { get; set; }

        /// <summary>
        /// Gets or sets the type for this object
        /// </summary>
        public XpoUrlObjectTypes ObjectType { get; set; }

        /// <summary>
        /// Gets or sets the text for this object
        /// </summary>
        public XpoUrlText Text { get; set; }

        /// <summary>
        /// Gets or sets the color for this object
        /// </summary>
        public XpoUrlColor Color { get; set; }

        /// <summary>
        /// Gets or sets the design for this object
        /// </summary>
        public XpoUrlDesign Design { get; set; }
    }
}